sphinx/config.py

```diff
@@ -278,9 +280,17 @@ class Config:
     def __init__(self, config: dict[str, Any] | None = None,
                  overrides: dict[str, Any] | None = None) -> None:
         raw_config: dict[str, Any] = config or {}
+        constants = raw_config.pop('__constants__', None) or frozenset()
+
+        # when empty, config.epub_copyright inherits from config.copyright, so
+        # treat it as constant if the latter is
+        if 'epub_copyright' not in raw_config and 'copyright' in constants:
+            constants |= frozenset(['epub_copyright'])
```

Comment:
For now, I would say that you should explicitly hardcode the allowed constants in a global variable (outside the Config class).